test/Mono.Linker.Tests.Cases/RequiresCapability/RequiresInCompilerGeneratedCode.cs

```diff
@@ -639,8 +639,6 @@ static async void TestReflectionAccess ()
 			{
 				LocalFunction ();
 
-				[ExpectedWarning ("IL2026")]
```

Comment:
Why... How?
This code SHOULD produce a warning in the linker (probably not in analyzer yet, I don't know).

Similar to all the other expected warnings removed in this file.

Reply:
Yeah, sorry, this one's wrong, I was disabling things to narrow down bugs and forgot to re-enable. I'll reset and file bugs against the analyzer if it's wrong

Reply:
Actually, I don't understand some of these. This one expects IL2026:

```
[RequiresUnreferencedCode ("Suppress in body")]
[RequiresAssemblyFiles ("Suppress in body")]
static async void TestDynamicallyAccessedMethod ()
{
	LocalFunction ();
	[ExpectedWarning ("IL2026")]
        void LocalFunction () => typeof (TypeWithMethodWithRequires).RequiresNonPublicMethods ();
}
```

But I don't understand why. The target isn't annotated with RUC, it has DAM.NonPublicMethods. Why should it produce IL2026?

Reply:
The `TypeWithMethodWithRequires` has a method which has RUC on it. The `RequiresNonPublicMethods` will apply the `[DAM(NonPublicMethods)]` on that type and thus will "mark" the method with the RUC -> thus the warning.